ioread.cpp

```diff
@@ -1453,3 +1464,52 @@ bool readFileCells(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
    exitOnError(success,"(READ_FILE_CELLS) Other error",MPI_COMM_WORLD);
    return success;
 }
+
+
+bool getFsgridDecomposition(vlsv::ParallelReader& file, std::array<int,3>& buffer){
```

Comment:
> I would make VLSVDIFF also use this function instead of having its own copy. 
This I tried to do. VLSVDIFF, however, uses a custom vlsvreaderinterface which was not at least trivially compatible here. This does lead to some code duplication, but at least the underlying call to FsGrid decomposition will not be duplicated.

> In fact, I would recommend having a separate call for readFsgridDecompositionFromVlsvFile which can return bool. If that fails, the program in question can then call computeFsGridDecomposition or computeLegacyFsGridDecomposition
Mh, originally I intended this to be the end-all function that could be called also from VLSVDIFF, so I wanted to avoid duplicating the logic... which obviously went just _ausgezeichnet_. Now, how robust is the base Reader class for calling from parallel vs sequential...?

Also, weren't we going to drop the legacy decomposition now?

Reply:
Legacy meaning calculation with that given looping and the same weights / handlings of dimensionalities, only adjusted to use integers. We need that one to be able to restart from old files / read old files for vlsvdiff. And for this, we need an override option.

The new decomposition should be inside fsgrid.hpp only, and will be the one we use for creating new internal fsgrids, and can be adusted later to give different preference for e.g. different dimensionalities.

Reply:
Yes, see last commits :). Both are in fsgrid.hpp, and it's calling for the legacy version when file does not have a decomposition. vlsvdiff update inbound.

Reply:
Also now, ofc, the new decomposition can, again, be however.